src/Planners/include/BipedalLocomotion/Planners/UnicyclePlanner.h

```diff
@@ -0,0 +1,120 @@
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_PLANNERS_UNICYCLE_PLANNER_H
+#define BIPEDAL_LOCOMOTION_PLANNERS_UNICYCLE_PLANNER_H
+
+#include <memory>
+#include <tuple>
+
+#include <BipedalLocomotion/Contacts/ContactList.h>
+#include <BipedalLocomotion/ParametersHandler/IParametersHandler.h>
+#include <BipedalLocomotion/System/Advanceable.h>
+
+namespace BipedalLocomotion::Planners
+{
+struct UnicycleKnot;
+class UnicyclePlanner;
+struct UnicyclePlannerState;
+struct UnicyclePlannerInput;
+} // namespace BipedalLocomotion::Planners
+
+struct BipedalLocomotion::Planners::UnicycleKnot
+{
+    UnicycleKnot(const double x = 0.0,
+                 const double y = 0.0,
+                 const double dx = 0.0,
+                 const double dy = 0.0,
+                 const double t = 0.0)
+        : x(x)
+        , y(y)
+        , dx(dx)
+        , dy(dy)
+        , time(t)
+    {
+    }
+
+    UnicycleKnot(const std::tuple<double, double>& position = {0, 0},
+                 const std::tuple<double, double>& velocity = {0, 0},
+                 const double time = 0.0)
+        : x(std::get<0>(position))
+        , y(std::get<1>(position))
+        , dx(std::get<0>(velocity))
+        , dy(std::get<1>(velocity))
+        , time(time)
+    {
+    }
+
+    bool operator==(const UnicycleKnot& rhs)
+    {
+        return this->x == rhs.x && this->y == rhs.y && this->dx == rhs.dx && this->dy == rhs.dy
+               && this->time == rhs.time;
+    }
+
+    double x;
+    double y;
+
+    double dx = 0.0;
+    double dy = 0.0;
+
+    double time = 0.0;
+};
+
+struct BipedalLocomotion::Planners::UnicyclePlannerInput
+{
+    UnicyclePlannerInput(const std::vector<UnicycleKnot>& knots,
+                         const double tf = 0.0,
+                         const double t0 = 0.0)
+        : t0(t0)
+        , tf(tf)
+        , knots(knots)
+    {
+    }
+
+    double t0;
+    double tf;
+
+    std::vector<UnicycleKnot> knots;
+};
+
+struct BipedalLocomotion::Planners::UnicyclePlannerState
+{
+    UnicyclePlannerState(const Contacts::ContactList& left = {},
+                         const Contacts::ContactList& right = {})
+        : left(left)
+        , right(right)
+    {
+    }
+
+    Contacts::ContactList left;
+    Contacts::ContactList right;
+};
+
+class BipedalLocomotion::Planners::UnicyclePlanner final
+    : public System::Advanceable<UnicyclePlannerInput, UnicyclePlannerState>
+{
+public:
+    UnicyclePlanner();
+
+    virtual ~UnicyclePlanner();
+
+    bool initialize(std::weak_ptr<const ParametersHandler::IParametersHandler> handler) override;
+
+    const UnicyclePlannerState& getOutput() const override;
+
+    bool isOutputValid() const override;
+
+    bool setInput(const UnicyclePlannerInput& input) override;
+
+    bool advance() override;
+
+private:
+    class Impl;
+    std::unique_ptr<Impl> pImpl;
```

Comment:
Sure! My opinion is that without private implementation, `m_` adds a context to the C++, while with private implementation, since there is just a single entry point, using `m_pImpl` is a bit ugly to look. Just personal taste tho, I'll align the naming!

Reply:
Done in 93ddd01